precompile/allow_list.go

```diff
@@ -70,44 +72,25 @@ func (c *AllowListConfig) Equal(other *AllowListConfig) bool {
 	return true
 }
 
-// Valid returns true iff [s] represents a valid role.
-func (s AllowListRole) Valid() bool {
-	switch s {
-	case AllowListNoRole, AllowListEnabled, AllowListAdmin:
-		return true
-	default:
-		return false
+// Verify returns an error if there is an overlapping address between admin and enabled roles
+func (c *AllowListConfig) Verify() error {
+	// check if both lists are empty
+	if len(c.EnabledAddresses) == 0 || len(c.AllowListAdmins) == 0 {
+		return nil
 	}
-}
-
-// IsNoRole returns true if [s] indicates no specific role.
-func (s AllowListRole) IsNoRole() bool {
-	switch s {
-	case AllowListNoRole:
-		return true
-	default:
-		return false
+	enabledMap := make(map[common.Address]struct{})
+	for _, enabledAddr := range c.EnabledAddresses {
+		if _, ok := enabledMap[enabledAddr]; !ok {
+			enabledMap[enabledAddr] = struct{}{}
```

Comment:
Can we ensure that there are no repeats within either list as well?

Reply:
hm that wouldn't be back-compatible if there are any chains activated a precompile with duplicate admin addresses.